syntax = "proto3";

//// https://protobuf.dev/reference/cpp/cpp-generated/#message
//option optimize_for = CODE_SIZE;

import "stepdirectory.proto";
import "miscinfofile.proto";
import "matrixfile.proto";
import "standardfontsfile.proto";
import "symbolsdirectory.proto";
import "attrlistfile.proto";

package Odb.Lib.Protobuf;

//
//  FileArchive.h
//

message FileArchive {

    map<string, StepDirectory> stepsByName = 1;
    optional MiscInfoFile miscInfoFile = 2;
    optional MatrixFile matrixFile = 3;
    optional StandardFontsFile standardFontsFile = 4;
    map<string, SymbolsDirectory> symbolsDirectoriesByName = 5;
    optional AttrListFile miscAttrListFile = 6;
    optional string productName = 7;
    optional string fileName = 8;
        
}